copy.go

```diff
@@ -112,6 +112,64 @@ type CopyGraphOptions struct {
 	FindSuccessors func(ctx context.Context, fetcher content.Fetcher, desc ocispec.Descriptor) ([]ocispec.Descriptor, error)
 }
 
+// WithMount enabled cross repository blob mounting.
+// sourceReference is the repository to use for mounting (the mount point).
+// mounter is the destination for the mount (a well-known implementation of this is *registry.Repository representing the target).
+// onMounted is called (if provided) when the blob is mounted.
+// The original PreCopy hook is called only on copy, and therefore not when the blob is mounted.
+func (opts *CopyGraphOptions) WithMount(sourceRepository string, mounter registry.Mounter, onMounted func(context.Context, ocispec.Descriptor)) {
```

Comment:
@Wwwsylvia Should `onMounted` return an `error` like PreCopy and PostCopy?

Reply:
I think it should for extendibility. But I don't have a good use case in mind though.

Reply:
`onMounted` now returns an error.